openedx/core/djangoapps/user_api/management/commands/email_opt_in_list.py

```diff
@@ -237,7 +237,7 @@ def _write_email_opt_in_prefs(self, file_handle, org_aliases, courses):
             email, full_name, course_id, is_opted_in, pref_set_datetime = row
             writer.writerow({
                 "email": email.encode('utf-8'),
-                "full_name": full_name.encode('utf-8'),
+                "full_name": full_name.encode('utf-8') if full_name else '',
                 "course_id": course_id.encode('utf-8'),
```

Comment:
This is a tricky one. It isn't supposed to be the case that there are some users without fullnames. Can you actually add a comment that we only need this line because of [ECOM-1995](https://openedx.atlassian.net/browse/ECOM-1995)?